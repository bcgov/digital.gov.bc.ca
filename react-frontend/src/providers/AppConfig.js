import React, { Component } from 'react';

export const AppConfigContext = React.createContext();

export class AppConfig extends Component {
  // TODO: It may be best to move the config set state into
  // constructor to make sure the config is assigned
  constructor() {
    super();
    this.state = { strapiApiUrl: null };
  }

  componentDidMount() {
    const xhr = new XMLHttpRequest();
    xhr.open('GET', '/config.json');

    xhr.send();

    xhr.onload = () => {
      if (xhr.status !== 200) {
        // analyze HTTP status of the xhrs response
        // do something
      } else {
        this.setState(JSON.parse(xhr.response));
      }
    };
  }

  render() {
    return (
      <AppConfigContext.Provider value={{ state: this.state }}>
        {this.props.children}
      </AppConfigContext.Provider>
    );
  }
}
