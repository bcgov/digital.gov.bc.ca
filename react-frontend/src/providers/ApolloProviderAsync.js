import React, { useContext, useEffect, useState } from 'react';
import { ApolloProvider } from 'react-apollo';
import ApolloClient from 'apollo-boost';

import { AppConfigContext } from './AppConfig';

// The apollo provider implements the strapi configuration
// by pulling the backend url from a config file. This allows
// the deployment config file instead of being baked into
// the build config.

function ApolloProviderAsync(props) {
  const config = useContext(AppConfigContext);
  const strapiURL = config['state']['strapiApiUrl'];

  // using client as state prevents it from being initialized needlessly
  const [client, setClient] = useState(
    new ApolloClient({
      uri: 'http://localhost:3000/graphql',
    })
  );

  // The useEffect is only called when config changes
  useEffect(() => {
    if (strapiURL) {
      setClient(new ApolloClient({ uri: strapiURL }));
    }
  }, [config, strapiURL]);

  return <ApolloProvider client={client}>{props.children}</ApolloProvider>;
}

export default ApolloProviderAsync;
