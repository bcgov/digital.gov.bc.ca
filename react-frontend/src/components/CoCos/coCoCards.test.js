import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CoCoCards from './coCoCards';
import { AppConfig } from '../../providers/AppConfig';
import { ApolloProvider } from 'react-apollo';
import ApolloClient from 'apollo-boost';
import { MockedProvider } from '@apollo/react-testing';

const originalError = console.error;

beforeAll(() => {
  console.error = (...args: any[]) => {
    if (/Warning.*not wrapped in act/.test(args[0])) {
      return;
    }

    originalError.call(console, ...args);
  };
});

afterAll(() => {
  console.error = originalError;
});

afterEach(cleanup);

it('Renders without crashing if not props are passed in', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <AppConfig>
      <MockedProvider>
        <CoCoCards></CoCoCards>
      </MockedProvider>
    </AppConfig>,
    div
  );
});
