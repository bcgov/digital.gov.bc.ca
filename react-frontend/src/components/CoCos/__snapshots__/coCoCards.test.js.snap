// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing if not props are passed in 1`] = `
<div
  className="sc-bXTfsO sc-dFtzdE esjtFX hckWov horizontalAlignment contentBlockContainer container"
  id="standards"
>
  <div
    className="row"
  >
    <div
      className="col-sm-12"
    >
      <h2
        className="sc-jWUzTF CITuv resourceHeading"
      >
        Explore
      </h2>
    </div>
  </div>
  <p>
    Loading...
  </p>
  <div
    className="row"
  >
    <div
      className="col-sm-12"
    >
      <h2
        className="sc-jWUzTF CITuv resourceHeading"
      >
        Support
      </h2>
      <p
        className="sc-jJoQpE fvraOj digitalParagraph"
      >
        Contact the Common Components team:
         
        <a
          className="sc-fKVsgm gjNLZe externalLink"
          href="mailto:digital.government@gov.bc.ca"
        >
          Digital.Government@gov.bc.ca
        </a>
        .
      </p>
    </div>
  </div>
</div>
`;
